### PR TITLE
Add kaniko task 0.5

### PR DESCRIPTION
**Pull request by pvlltvk**
Task now uses --digest-file flag to create a results file with the
digest of a built image.

# Changes

Task now uses `--digest-file` flag to create a results file with the digest of a built image as was discussed in #324 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/kaniko/0.5/kaniko.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: kaniko
+  labels:
+    app.kubernetes.io/version: "0.5"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/tags: image-build
+    tekton.dev/displayName: "Build and upload container image using Kaniko"
```
Comment (piyush-garg) [inline task/kaniko/0.5/kaniko.yaml:10]: can you please add category and platform annotation

Comment (pvlltvk) [reply task/kaniko/0.5/kaniko.yaml:11]: Done, thank you!

**Review by vinamra28 — COMMENTED** [review]
PR looks good. Can you please squash the commits as well?

**Comment by PuneetPunamiya** [comment]
@pvlltvk  Could you please squash the commits into a single commit ??

**Comment by pvlltvk** [comment]
@PuneetPunamiya @vinamra28 Hi, I've just squashed my commits, take a look.

**Pull request merged**